Sources/Pokepay/Pokepay.swift

```diff
@@ -301,8 +301,15 @@ public struct Pokepay {
             }
         }
 
-        public func getAuthorizationUrl() -> String {
-            return "\(wwwBaseURL)/oauth/authorize?client_id=\(clientId)&response_type=code"
+        public func getAuthorizationUrl(contact: String? = nil) -> String {
+            let url = "\(wwwBaseURL)/oauth/authorize?client_id=\(clientId)&response_type=code"
+            if contact != nil {
+                let encodedContact = contact!.addingPercentEncoding(withAllowedCharacters: NSCharacterSet.urlQueryAllowed)
+                precondition(encodedContact != nil)
+                return "\(url)&contact=\(encodedContact!)"
+            } else {
+                return url
+            }
```

Comment:
[should] Checking `Optional` value with `if` can be replaced with `guard ... else` which is thought to be better style in some documents for readability.

Like,

```swift
Using a guard statement for requirements improves the readability of your code, compared to doing the same check with an if statement. It lets you write the code that’s typically executed without wrapping it in an else block, and it lets you keep the code that handles a violated requirement next to the requirement.
```

at [Early Exits of the official Swift language guide](https://docs.swift.org/swift-book/LanguageGuide/ControlFlow.html#ID525),  or,

```swift
guards for Early Exits

A guard statement, compared to an if statement with an inverted condition, provides visual emphasis that the condition being tested is a special case that causes early exit from the enclosing scope.
```

in [Swift Style Guide](https://google.github.io/swift/#guards-for-early-exits) by Google.


```suggestion
            guard let contact = contact else {
                return url
            }

            let encodedContact = contact.addingPercentEncoding(withAllowedCharacters: NSCharacterSet.urlQueryAllowed)
            precondition(encodedContact != nil)
            return "\(url)&contact=\(encodedContact!)"
```